src/main/java/com/arangodb/entity/arangosearch/analyzer/NormAnalyzerProperties.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * DISCLAIMER
+ *
+ * Copyright 2016 ArangoDB GmbH, Cologne, Germany
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright holder is ArangoDB GmbH, Cologne, Germany
+ */
+
+package com.arangodb.entity.arangosearch.analyzer;
+
+
+import com.arangodb.velocypack.annotations.SerializedName;
+
+import java.util.Objects;
+
+/**
+ * @author Michele Rastelli
+ */
+public class NormAnalyzerProperties {
+
+    /**
+     * TODO: clarify: what are the supported locales? can this field be an enum?
```

Comment:
Not really. It can be any supported language code, optionally followed by a country code, which may or may not make a difference - I don't think that Snowball (the stemming library) does anything with this information, but ICU (the tokenization/normalization library) might.

Then the coding should follow, and it should generally be set to `utf-8`, because inside ArangoDB we use normalized UTF-8 strings throughout AFAIK. If you omit it here then it is very likely that it mangles the input because it will interpret it in a wrong way or some other code later may decode it into something unusable. It is optional in the Analyzer configuration nonetheless, and if you exclusive use ASCII characters in strings then you won't see negative effects.

You can optionally provide a locale variant string as last piece for the locale value, and this is highly language dependent: For German, there is a phone book variant for instance, which sorts strings differently, namely umlauts ä/ö/ü are treated like ae/oe/ue instead of as a/o/u. The availability of such variants depends on the ICU data file. The subset we current ship does not feature the German phone book encoding for instance, but the full version does. This extends to whole languages even, e.g. Icelandic is only available in the full ICU version. The user may replace the bundled data file with the full one, which should enable correct sorting of Icelandic strings etc., but I'm not sure if this applies to ArangoSearch as well or if it is independent / requires something compiled into the binary to support the additional features. See https://github.com/arangodb/arangodb/issues/7111